lib/brexit_checker/results_audiences.rb

```diff
@@ -0,0 +1,56 @@
+class BrexitChecker::ResultsAudiences
+  class << self
+    def populate_business_groups(audience_actions, selected_criteria)
+      return [] if audience_actions.blank? || selected_criteria.blank?
+
+      {
+        actions: audience_actions,
+        criteria: filter_criteria_by_actions(audience_actions, selected_criteria),
+      }
+    end
+
+    def populate_citizen_groups(audience_actions, selected_criteria)
+      return [] if audience_actions.blank? || selected_criteria.blank?
+
+      all_possible_groups = BrexitChecker::Groups.load_all
+      grouped_actions = all_possible_groups.each_with_index.map do |group, group_index|
+        actions_in_group = filter_actions_by_group(audience_actions, group.key)
+        if actions_in_group.empty?
+          nil
+        else
+          {
+            group: group,
+            group_index: group_index + 1,
+            actions: actions_in_group,
+            criteria: filter_criteria_by_actions(actions_in_group, selected_criteria),
+          }
+        end
+      end
+      grouped_actions.compact
+    end
+
+  private
+
+    def filter_actions_by_group(actions, group_key)
+      actions.select { |action| action.grouping_criteria.include?(group_key) }
+    end
+
+    def extract_groupings(actions)
```

Comment:
I think this is unused

Reply:
great catch, I really need to spend some time on my IDE to catch these in ruby

Reply:
removed with squash and push shortly